fhir-term-graph/src/main/java/com/ibm/fhir/term/graph/loader/impl/CodeSystemTermGraphLoader.java

```diff
@@ -0,0 +1,225 @@
+        }
+
+        g.tx().commit();
+    }
+
+    private void createConceptVertices() {
+        for (Concept concept : concepts) {
+            String code = concept.getCode().getValue();
+            Vertex conceptVertex = g.addV("Concept")
+                    .property("code", code)
+                    .property("codeLowerCase", normalize(code))
+                    .next();
+
+            if (concept.getDisplay() != null) {
+                String display = concept.getDisplay().getValue();
+                g.V(conceptVertex)
+                    .property("display", display)
+                    .next();
+            }
+
+            for (Designation designation : concept.getDesignation()) {
+                Vertex designationVertex = g.addV("Designation")
+                        .property("value", designation.getValue().getValue())
+                        .next();
+
+                if (designation.getUse() != null) {
+                    g.V(designationVertex)
+                        .property("use", designation.getUse().getCode().getValue())
+                        .next();
+                }
+
+                if (designation.getLanguage() != null) {
+                    g.V(designationVertex)
+                        .property("language", designation.getLanguage().getValue())
+                        .next();
+                }
+
+                g.V(conceptVertex).addE("designation").to(designationVertex).next();
+            }
+
+            for (Property property : concept.getProperty()) {
+                Element value = property.getValue();
+
+                String key = "value" + value.getClass().getSimpleName();
+
+                Vertex propertyVertex = g.addV("Property_")
+                        .property("code", property.getCode().getValue())
+                        .property(key, toObject(value))
+                        .next();
+
+                if (value.is(DateTime.class)) {
+                    g.V(propertyVertex).property("valueDateTimeLong", toLong(value.as(DateTime.class))).next();
+                } else if (value.is(Decimal.class)) {
+                    g.V(propertyVertex).property("valueDecimalString", value.as(Decimal.class).getValue().toPlainString()).next();
+                }
+
+                g.V(conceptVertex).addE("property_").to(propertyVertex).next();
+            }
+
+            g.V(codeSystemVertex).addE("concept").to(conceptVertex).next();
+
+            conceptVertexMap.put(concept, conceptVertex);
+        }
+
+        g.tx().commit();
+    }
+
+    private void createEdges() {
+        for (Concept concept : concepts) {
+            Vertex v = conceptVertexMap.get(concept);
+            for (Concept child : concept.getConcept()) {
+                Vertex w = conceptVertexMap.get(child);
+                g.V(w).addE(FHIRTermGraph.IS_A).to(v).next();
+            }
+        }
+
+        g.tx().commit();
+    }
+
+    public static void main(String[] args) throws Exception {
```

Comment:
should this be in a tools package?   or as is in the impl   package is fine?

Reply:
Each Loader type has its own "main" so they can be run/tested independently. The entry point that we'd externalize, however, is FHIRTermGraphLoaderMain which is in a `main` package.